### PR TITLE
Developer documentation: Several small improvements

### PR DESCRIPTION
**Pull request by Felix-Dev**
Reading through the developer documentation, I noticed several typos and felt that the reading flow could be improved here and there. As such, this PR aims to:

* Fix typos
* Improve the reading flow
* Make links language agnostic

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -151,33 +150,30 @@ public void TestCleanup()
...
 
-One of the core values of WinUI is the way that it brings controls to a wide 
-variety of versions of Windows, handling version compatibility differences so 
-that developers using WinUI don't have to. As such, testing WinUI changes on 
-different versions is sometimes necessary. To accomplish this you will need to 
-at times set up older versions of Windows for testing. To get these earlier 
-versions you can make use of a Visual Studio subscription [as described here](https://docs.microsoft.com/azure/virtual-machines/windows/client-images).
+One of the core features of WinUI is that it maintains down-level compatibility with earlier versions of Windows 10, handling version compatibility differences so 
+that developers using WinUI don't have to. As such, testing your changes on 
+different Windows versions is sometimes necessary. Windows versions can be acquired using a Visual Studio subscription [as described here](https://docs.microsoft.com/azure/virtual-machines/windows/client-images), if required.
```
Comment (chingucoding) [inline docs/developer_guide.md:162]: @ranjeshj Are there ways to get a VM if you don't have any of the subscriptions listed here?

File: docs/developer_guide.md [hunk 2]
```diff
@@ -30,43 +29,43 @@ While WinUI is designed to work against many versions of Windows, you will need
...
 all the boxes when prompted), or you can manually download them from here: 
-https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk
+https://developer.microsoft.com/windows/downloads/windows-10-sdk
 
 
 ## Building the repository
 
-Building the solution **MUXControls.sln** will build all projects. 
-Generally you will want to set your configuration to **Debug**, **x64**, and 
+The **MUXControls.sln** solution is used to build all projects/controls. The **MUXControlsInnerLoop.sln** solution can be used to build only select projects/controls.
```
[the remark this answers is not in the document]
Comment (Felix-Dev) [reply docs/developer_guide.md:37]: What do you think of this?
`
The MUXControls.sln solution is used to build all feature areas (such as controls, layout panels and composition elements). The **MUXControlsInnerLoop.sln** solution can be used to build only select feature areas.
`

Mentioning composition elements to capture things like brushes (acrylic, reveal, radial gradient,...), Light APIs, the `Effects` project,... and as an explicit mention here already preparing for WinUI 3 which will, broadly speaking, make available controls, input and *composition* APIs outside of the OS.

While more WinUI 2 components could be listed in the enumeration (like Converters) this is not meant as an exhaustive list but rather to give the WinUI-novice an understanding what we mean we "feature areas" here. Those three areas seem like good candidates for now.

[the remark this answers is not in the document]
Comment (chingucoding) [reply docs/developer_guide.md:37]: That sounds great! That list is good to show what we understand as feature area in this context, while also not overwhelming users with too many examples.

[the remark this answers is not in the document]
Comment (Felix-Dev) [reply docs/developer_guide.md]: Done.

File: docs/developer_guide.md (hunk 1 above)
Comment (StephenLPeters) [reply docs/developer_guide.md:162]: I don't think so :(

---
In reply to: [485197625](https://github.com/microsoft/microsoft-ui-xaml/pull/3262#discussion_r485197625) [](ancestors = 485197625)

**Review by StephenLPeters — APPROVED** [review]
:shipit:

**Comment by StephenLPeters** [comment]
@anawishnoff do you want to take a look?

**Review by anawishnoff — APPROVED** [review]

**Comment by anawishnoff** [comment]
Thanks @Felix-Dev! This looks great.

**Pull request merged**